WelsonJS.Toolkit/WelsonJS.Launcher/AssemblyLoader.cs

```diff
@@ -359,6 +365,54 @@ private static void DownloadFile(string url, string dest)
         }
 
 
+        private static bool TryDownloadGzipToFile(string gzUrl, string dest)
+        {
+            string tempFile = dest + ".tmp";
+
+            try
+            {
+                using (var res = Http.GetAsync(gzUrl).GetAwaiter().GetResult())
+                {
+                    if (res.StatusCode == HttpStatusCode.NotFound)
+                        return false;
+
+                    res.EnsureSuccessStatusCode();
+
+                    using (Stream s = res.Content.ReadAsStreamAsync().GetAwaiter().GetResult())
+                    using (var gz = new GZipStream(s, CompressionMode.Decompress))
+                    using (var fs = new FileStream(tempFile, FileMode.Create, FileAccess.Write))
+                    {
+                        gz.CopyTo(fs);
+                    }
+
+                    File.Move(tempFile, dest);
+
+                    return true;
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger?.Warn("Failed to download or decompress gzipped file from {0}: {1}", gzUrl, ex.Message);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Input is being passed directly to log statements, which could allow attackers to inject malicious content into log files. This can lead to log forging, data integrity issues, or even remote code execution in some cases. Sanitize all external input before logging, especially data from HTTP requests. Use encoding functions like HtmlEncode() or validated libraries to properly sanitize the input. Learn more - https://cwe.mitre.org/data/definitions/117.html